diesel/src/types/mod.rs

```diff
@@ -1,720 +1,109 @@
-#[postgres(oid = "1082", array_oid = "1182")]
-#[sqlite_type = "Text"]
-#[mysql_type = "Date"]
-pub struct Date;
-
-/// The interval SQL type.
-///
-/// This type is currently only implemented for PostgreSQL.
-///
-/// ### [`ToSql`](trait.ToSql.html) impls
-///
-/// - [`PgInterval`] which can be constructed using [`IntervalDsl`]
-///
-/// ### [`FromSql`](trait.FromSql.html) impls
-///
-/// - [`PgInterval`] which can be constructed using [`IntervalDsl`]
-///
-/// [`PgInterval`]: ../pg/data_types/struct.PgInterval.html
-/// [`IntervalDsl`]: ../pg/expression/extensions/trait.IntervalDsl.html
-#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
-#[postgres(oid = "1186", array_oid = "1187")]
-pub struct Interval;
-
-/// The time SQL type.
-///
-/// ### [`ToSql`](trait.ToSql.html) impls
-///
-/// - [`chrono::NaiveTime`][NaiveTime] with `feature = "chrono"`
-///
-/// ### [`FromSql`](trait.FromSql.html) impls
-///
-/// - [`chrono::NaiveTime`][NaiveTime] with `feature = "chrono"`
-///
-/// [NaiveTime]: /chrono/naive/time/struct.NaiveTime.html
-#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
-#[postgres(oid = "1083", array_oid = "1183")]
-#[sqlite_type = "Text"]
-#[mysql_type = "Time"]
-pub struct Time;
-
-/// The timestamp SQL type.
-///
-/// ### [`ToSql`](trait.ToSql.html) impls
-///
-/// - [`std::time::SystemTime`][SystemTime] (PG only)
-/// - [`chrono::NaiveDateTime`][NaiveDateTime] with `feature = "chrono"`
-/// - [`time::Timespec`][Timespec] with `feature = "deprecated-time"` (PG only)
-///
-/// ### [`FromSql`](trait.FromSql.html) impls
-///
-/// - [`std::time::SystemTime`][SystemTime] (PG only)
-/// - [`chrono::NaiveDateTime`][NaiveDateTime] with `feature = "chrono"`
-/// - [`time::Timespec`][Timespec] with `feature = "deprecated-time"` (PG only)
-///
-/// [SystemTime]: https://doc.rust-lang.org/nightly/std/time/struct.SystemTime.html
-/// [NaiveDateTime]: /chrono/naive/datetime/struct.NaiveDateTime.html
-/// [Timespec]: /time/struct.Timespec.html
-#[derive(Debug, Clone, Copy, Default, QueryId, SqlType)]
-#[postgres(oid = "1114", array_oid = "1115")]
-#[sqlite_type = "Text"]
-#[mysql_type = "Timestamp"]
-pub struct Timestamp;
-
-/// The nullable SQL type.
-///
-/// This wraps another SQL type to indicate that it can be null.
-/// By default all values are assumed to be `NOT NULL`.
-///
-/// ### [`ToSql`](trait.ToSql.html) impls
-///
-/// - Any `T` which implements `ToSql<ST>`
-/// - `Option<T>` for any `T` which implements `ToSql<ST>`
-///
-/// ### [`FromSql`](trait.FromSql.html) impls
-///
-/// - `Option<T>` for any `T` which implements `FromSql<ST>`
-#[derive(Debug, Clone, Copy, Default)]
-pub struct Nullable<ST: NotNull>(ST);
-
-#[cfg(feature = "postgres")]
-pub use pg::types::sql_types::*;
+#[deprecated(since = "1.1.0", note = "Use `sql_types::Timestamp` instead")]
+pub type Timestamp = ::sql_types::Timestamp;
 
+#[deprecated(since = "1.1.0", note = "Use `sql_types::Datetime` instead")]
 #[cfg(feature = "mysql")]
```

Comment:
The types in `sql_types` don't have this cfg's anymore, so maybe skip them here too?

Reply:
The `Datetime` type is a MySQL specific type, and is only present when the `mysql` feature is enabled.